deployment/helm/crds/akri-configuration-crd.yaml

```diff
@@ -18,6 +18,17 @@ spec:
                 protocol: # {{ProtocolHandler}}
                   type: object
                   properties:
+                    zeroconf: # {{ZeroconfDiscoveryHandler}}
+                      type: object
+                      properties:
+                        kind:
+                          type: string
+                        name:
+                          type: string
```

Comment:
would making these string[] make sense?  would someone want to deploy one broker to handle zeroconf devices with several names/domains?

for onvif, i found having the ability to exclude to be helpful as well.  i can imagine wanting your broker to handle every device of kind=foo except ones in domain=blah.

Reply:
maybe a vNext sort of thing?

Reply:
I was going for simple(st).

Assuming we have 6 services (on `local`):
```console
a _http._tcp 7777 local project=akri protocol=x location=foo # HTTP
b _rust._tcp 8888 local project=akri protocol=x location=bar
c _rust._tcp 9999 local project=akri protocol=x location=baz

d _rust._tcp 7777 local project=akri protocol=y location=foo
e _rust._tcp 8888 local project=akri protocol=y location=bar
f _http._tcp 9999 local project=akri protocol=y location=baz # HTTP
```

I propose (!?):

For `protocol=x` we'd deploy one configuration:

```YAML
spec:
  protocol:
    zeroconf:
      kind: "_rust._tcp"
      domain: "local"
      txtRecords:
        project: akri
        protocol: x
```

... and for `protocol=y` another

And, because, in the above, we're filtering by `kind=_rust._tcp`, we'd only see 2 services on each Instance (`_http._tcp` would be excluded).

And, because the Instance includes an environment variable for each TXT record, within any `protocol=[x|y]`, we'd be able to infer location `AKRI_ZEROCONF_DEVICE_LOCATION=[foo|bar|baz]`.

What do you think?

Reply:
Simple seems fine.  It isn't clear to me whether people would prefer to have 1 configuration handle a lot of instances or if they want a closer to 1:1 relationship between configurations and instances.  We can adjust if usage dictates a different approach. :)